.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,57 @@
+name: build and validate
+on:
+  push:
+    branches:
+      - main
+  pull_request_target:
+    branches: 
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+       # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          fetch-depth: 0
+      
+      - name: create checksum file
+        uses: hypertrace/github-actions/checksum@main
+
+      - name: Cache packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle
+          key: gradle-packages-${{ runner.os }}-${{ github.job }}-${{ hashFiles('**/checksum.txt') }}
+          restore-keys: |
+            gradle-packages-${{ runner.os }}-${{ github.job }}
+            gradle-packages-${{ runner.os }}
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_READ_USER }}
+          password: ${{ secrets.DOCKERHUB_READ_TOKEN }}
+      
+      - name: Build with Gradle
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: build dockerBuildImages
+
+  validate-helm-charts:
```

Comment:
This can move into the test workflow, no secrets are used requiring p_r_target

Reply:
yes but there are no tests here so kept in the same file as we are keeping it in build and validate workflow at other places.

Reply:
Given that the workflows are divided by triggers though,
- Would it make sense to merge the docker workflow into the other
- In other places, I would move it too - p_r_target is a last resort for a couple different reasons (less secure, difficult to change since not workflow changes aren't testable without merging)

Reply:
- we shouldn't merge docker workflow into other as I will be moving that to pull_request separately. 
- I can create separate wokflow for that now and then I will merge that with e2e workflow when I will handle that. 
- or we can keep it as it is now and then handle with e2e solution.

Reply:
Unless you feel strongly, my preference would be to create the separate p_r based workflow now since that's a different issue than the e2e test one. If we want to leave the e2e as its own workflow pending the resolution around tags, that's fine.

Another thought about the e2e solution though - it's currently not doing a docker login before building the image, unlike in the `build` job. So...
 - Accept that it might get rate limited, unless we know of some reason why it wouldn't. In this case, we might as well remove the login from build and put that in a p_r triggered workflow too.
 - Keep it in a p_r_target workflow (and maybe just replace the build job since it appears to be a superset of it)
 - Migrate our base images to GHCR and not worry about rate limiting. This seems ideal, but is a separate piece of work most likely (but if its our go forward strategy, accepting the temporary rate limiting may be acceptable)

Reply:
Okay. will convert build workflow to pull_request and remove creds as this repo won't be getting lot of PRs anyways. Let's plan the base image migration to GHCR once this Github actions migration is completed

Reply:
made similar change to all data service repos.